libbeat/cmd/instance/beat.go

```diff
@@ -285,6 +286,226 @@ func NewBeat(name, indexPrefix, v string, elasticLicensed bool, initFuncs []func
 	return &Beat{Beat: b}, nil
 }
 
+// NewBeatReceiver creates a Beat that will be used in the context of an otel receiver
+func NewBeatReceiver(settings Settings, receiverConfig map[string]interface{}, consumer consumer.Logs) (*Beat, error) {
+	b, err := NewBeat(settings.Name,
+		settings.IndexPrefix,
+		settings.Version,
+		settings.ElasticLicensed,
+		settings.Initialize)
+	if err != nil {
+		return nil, err
+	}
+
+	b.Info.LogConsumer = consumer
+
+	// begin code similar to configure
+	if err = plugin.Initialize(); err != nil {
+		return nil, fmt.Errorf("error initializing plugins: %w", err)
+	}
+
+	b.InputQueueSize = settings.InputQueueSize
+
+	cfOpts := []ucfg.Option{
+		ucfg.PathSep("."),
+		ucfg.ResolveEnv,
+		ucfg.VarExp,
+	}
+
+	tmp, err := ucfg.NewFrom(receiverConfig, cfOpts...)
+	if err != nil {
+		return nil, fmt.Errorf("error converting to ucfg: %w", err)
```

Comment:
[Suggestion]

I believe we can add a little bit more context here
```suggestion
		return nil, fmt.Errorf("error converting receiver config to ucfg: %w", err)
```